test/test_serialization.py

```diff
@@ -58,3 +60,32 @@ def test_default_component_from_dict_unregistered_component(request):
 
     with pytest.raises(DeserializationError, match=f"Class '{component_name}' can't be deserialized as 'Mock'"):
         default_from_dict(Mock, {"type": component_name})
+
+
+def test_from_dict_import_type():
+    pipeline_dict = {
+        "metadata": {},
+        "max_loops_allowed": 100,
+        "components": {
+            "greeter": {
+                "type": "sample_components.greet.Greet",
+                "init_parameters": {
+                    "message": "\nGreeting component says: Hi! The value is {value}\n",
+                    "log_level": "INFO",
+                },
+            }
+        },
+        "connections": [],
+    }
+
+    # remove the target component from the registry if already there
+    component.registry.pop("sample_components.greet.Greet", None)
+    # remove the module from sys.modules if already there
+    sys.modules.pop("sample_components.greet", None)
+
+    p = Pipeline.from_dict(pipeline_dict)
+
+    from sample_components.greet import Greet
```

Comment:
Nitpick: `Greet` should be already imported (if I understood the code above). If so, we may add `if TYPE_CHECKING:` here to make it available only to IDEs and not to the test.
```suggestion
    if TYPE_CHECKING:
        from sample_components.greet import Greet
```

Reply:
That's not for type checking, I actually need the symbol `Greet` to compare with the output of `get_component`